nengo_loihi/simulator.py

```diff
@@ -226,6 +226,7 @@ def __del__(self):
     def __enter__(self):
         for sim in self.sims.values():
             sim.__enter__()
+        self._runner = StepRunner(self.model, self.sims, self.precompute, self.timers)
```

Comment:
Does this mean that the user has to use the simulator in a `with` block, and we no longer support something like:
```
sim = nengo_loihi.Simulator(net)
sim.run(1.0)
sim.close()
```

Reply:
That's true, this is something I hadn't considered. This is probably a conversation we should have team-wide, as right now the assumption is that constructing the simulator also "opens" it, `__enter__` does nothing, and `__exit__` calls `close`. If we want a two-step initialization, like it sounds like has some advantages in Nengo Loihi, then we need to add an `open` method that people need to call, which we should also add to all backends.

Reply:
Decision is to move the `StepRunner` creation to `__init__` and make a card for adding `open` methods for all backends for later.

Reply:
Done in 96bd3779